static/js/components/SpinnerButton.js

```diff
@@ -32,47 +32,42 @@ export default class SpinnerButton extends React.Component {
     }
   }
 
+  isDisabled = () => (this.props.disabled || this.props.spinning) ? true : undefined;
+
+  // wrap onClick to return undefined if disabled and to set recentlyClicked if clicked
+  onClickWrapper = e => (
+    this.isDisabled() ? undefined : (...args) => {
+      this.setState({
+        recentlyClicked: true
+      });
+      return this.props.onClick(...args);
+    }
+  );
```

Comment:
Sure, I assumed that your comment was moving the whole thing including the if statement, but this works too